packages/camera/lib/camera.dart

```diff
@@ -307,6 +307,9 @@ class CameraController extends ValueNotifier<CameraValue> {
   StreamSubscription<dynamic> _imageStreamSubscription;
   Completer<void> _creatingCompleter;
 
+  /// True after [CameraController.dispose] has completed successfully.
+  bool get isDisposed => _isDisposed;
```

Comment:
we typically expose things like this guarded by asserts, so they're only valid in debug mode.

This is to discourage people from adding checks like this to release mode code, where it should never really be necessary. 

IOW, making it easier to assert program correctness is great. Allowing someone to bog down code with 1000s of small checks that really should only be done during development is not as preferable.

Reply:
thank you for providing feedback. I agree with the idea on only allowing this during during development. However I am not really sure how to "expose things like this guarded by asserts"?

Does this imply we don't expose a property but rather add assert statements to the methods? I feel a bit stupid here :P, but I just cannot wrap my head around it.

Reply:
```suggestion
  /// Checks whether [CameraController.dispose] has completed successfully.
  ///
  /// This is a no-op when asserts are disabled.
  void debugDisposed() => assert(_isDisposed);
```

Reply:
Or maybe `debugCheckDisposed` or something.

Reply:
@bparrishMines what do you think? 

In general it would be bad to have any code that can potentially access the `CameraController` when it is already disposed. This means having the `debugCheckIsDisposed` around should be helpful for developers trying to debug their application, but should not be used in production. This will also prevent code constructs like this, which has high risk on state conflicts:

```dart
var cameraController = CameraController();

// do stuff with the camera controller
...

if (cameraController.isDisposed) {
  cameraController = CameraController();
}

// continue doing stuff with the camera controller
```

Reply:
Yea, that makes sense to avoid that type of pattern. I'm fine with going with a `debugCheckIsDisposed` instead.

Reply:
@dnfield thank you for clarifying the coding structure that was very helpful. I took your recommendations and updates the pull request. 

Note that I had to change:
```dart
void debugCheckIsDisposed() => assert(_isDisposed);
```
to:
```dart
void debugCheckIsDisposed() {
  assert(_isDisposed);
}
```

otherwise we would receive a compile error.

I would appreciate it if you could have another look.